CONTRIBUTING.md

```diff
@@ -0,0 +1,135 @@
+# Contributing to nidaqmx
+
+Contributions to **nidaqmx** are welcome from all!
+
+**nidaqmx** is managed via [git](https://git-scm.com), with the canonical upstream repository hosted
+on [GitHub](http://developercertificate.org/).
+
+**nidaqmx** follows a pull-request model for development.  If you wish to contribute, you will need
+to create a GitHub account, fork this project, push a branch with your changes to your project, and
+then submit a pull request.
+
+See [GitHub's official documentation](https://help.github.com/articles/using-pull-requests/) for
+more details.
+
+# Getting Started
+
+To contribute to this project, it is recommended that you follow these steps:
+
+1. Ensure you have [poetry](https://python-poetry.org/) [installed](https://python-poetry.org/docs/#installation)
+2. Fork the repository on GitHub.
+3. Install **nidaqmx** dependencies using ``poetry install``
+4. Run the unit tests on your system (see Testing section). At this point, if any tests fail, do not
+begin development. Try to investigate these failures. If you're unable to do so, report an issue
+through our [GitHub issues page](http://github.com/ni/nidaqmx-python/issues).
+5. Write new tests that demonstrate your bug or feature. Ensure that these new tests fail.
+6. Make your change.
+7. Run all the unit tests again (which include the tests you just added), and confirm that they all
+pass.
+8. Send a GitHub Pull Request to the main repository's master branch. GitHub Pull Requests are the
+expected method of code collaboration on this project.
+
+# Testing
+
+In order to be able to run the **nidaqmx** unit tests, your setup should meet the following minimum
+requirements:
+
+- Setup has a machine with NI-DAQmx or the NI-DAQmx Runtime installed.
+   - Currently the minimum supported NI-DAQmx version to run all tests is 21.3.
+- Machine has a supported version of CPython or PyPy installed.
+- Machine has [poetry](https://python-poetry.org/) installed.
+- Machine has an X-Series DAQ device connected to it (we ran the tests 
+  using a PCIe-6363 or a USB-6351).
+
+Before running any unit tests, an NI MAX configuration needs be imported. The MAX configuration
+simply contains some custom scales used during testing. The MAX configuration file is located at
+``nidaqmx\tests\max_config\nidaqmxMaxConfig.ini``. Refer to this [KB article](http://digital.ni.com/public.nsf/allkb/0E0D3D7C4AA8903886256B29000C9D5A)
+for details on how to import a MAX Configuration.
+
+To run the **nidaqmx** unit tests in a specific version of Python, run the following command in the
+root of the distribution:
+
+```sh
+$ poetry run pytest
+```
+
+To run the unit tests in all Python interpreters supported by **nidaqmx**, run the following
+commands in the root of the distribution:
+
+```sh
+$ pip install tox
```

Comment:
If you add tox to the dev tools I believe you should be able to call `poetry run tox`

Reply:
Yeah, that works (if I add `tox` as a dev dependency, like you described above). It is kind of weird though because then `tox` creates its own dev environments seeded from within the `poetry` dev environment. That isn't a big deal, though, and it makes it a bit more straight-forward. So, yeah, I'll do that :)

Reply:
Yeah, it is weird interaction but it is nice to have the tools used listed on the same spot for the most part.